x-pack/solutions/security/plugins/security_solution/public/flyout/document_details/right/components/graph_preview_container.tsx

```diff
@@ -72,6 +75,12 @@ export const GraphPreviewContainer: React.FC = () => {
     },
   });
 
+  if (hasGraphRepresentation) {
```

Comment:
does this mean that every time the container is rendered (and re-rendered), it fires a telemetry event? not sure if this is what you want?
in the flyout code, we usually emit an event when user navigate to the tab, in those cases, it will only count as one event despite having subsequent interaction in that tab

Reply:
to show what @christineweng commented on, here a video. Luckily we're memoizing a parent component so this component isn't re-rendered too many times, but still I'm seeing 2 renders every time the parent component is expanded

https://github.com/user-attachments/assets/7403e6e1-a541-40be-9af9-2b9c4099d4e4

Reply:
to add a bit more, you can see that the numbers collected are not going to reflect the intended `how many users ingested data that is graph compliant`. If users just navigate around the flyout without interacting with the graph at all, plenty of telemetry information will be logged...

https://github.com/user-attachments/assets/a8ba5798-7eae-4bf1-a300-acd3bab56a93

I think having something like the alert id logged would allow to filter the data and remove duplicate when we visualize it on dashboard or such?

Reply:
Thank you for the comments and the detailed screen sharing!
I refactored the code to improve the implementation
I used `useGeneratedHtmlId` to identify a unique rendering. Later I use `useEffect` that depends on the `renderingId` to report the telemetry